### DISCUSSION TITLE
Server Error when creating permission with constraints

### DISCUSSION THREAD

**Discussion by Kani999** [netbox-community/netbox#15019] [Help Wanted!]
### Deployment Type

Self-hosted

### NetBox Version

v3.7.1

### Python Version

3.8

### Steps to Reproduce

1. Add new Permission https://demo.netbox.dev/users/permissions/add/ 
2. Setup **Constraints** as `{"role": "server"}` (in API docs it should accept `slug` as param)
3. Create
4. Obtain Server Error
```
<class 'ValueError'>

Field 'id' expected a number but got 'server'.

Python version: 3.8.10
NetBox version: 3.7.1
Plugins: 
  netbox_demo: 0.3.1
```

### Expected Behavior

Server Error should not occur. It should display some kind of validation popup.

### Observed Behavior

Server Error occurs
```
<class 'ValueError'>

Field 'id' expected a number but got 'server'.

Python version: 3.8.10
NetBox version: 3.7.1
Plugins: 
  netbox_demo: 0.3.1
```

**Comment by jeremystretch** [comment]
You need to specify which attribute of the role to match, e.g. `name` or `slug`, when defining the constraint.

```
{"role__slug": "server"}
```

**Reply by Kani999** [reply]
That’s right, but even that it should not end with Server Error 500. It should be catch and validation error should be raised.
